nuxeo-core/nuxeo-core-api/src/test/java/org/nuxeo/ecm/core/DummyLoginAs.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2018 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.nuxeo.ecm.core;
+
+import java.security.Principal;
+
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+import org.nuxeo.ecm.core.api.impl.UserPrincipal;
+import org.nuxeo.ecm.core.api.local.ClientLoginModule;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+import org.nuxeo.runtime.api.login.LoginAs;
+
+/**
+ * Computation rely on Framework.loginAsUser(String username).
+ * <p>
+ * This Framework wrapper method relies on the LoginAs service, thus we have to provide an implementation that allows
+ * tests to be executed.
+ *
+ * @since HF
+ */
+public class DummyLoginAs implements LoginAs {
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public LoginContext loginAs(String username) throws LoginException {
+        boolean isAdministrator = SecurityConstants.ADMINISTRATOR.equalsIgnoreCase(username);
+        Principal principal = new UserPrincipal(username, null, false, isAdministrator);
+        ClientLoginModule.getThreadLocalLogin().push(principal, null, null);
+        return new LoginContext("nuxeo-client-login") {
+            @Override
+            public void logout() throws LoginException {
+                ClientLoginModule.getThreadLocalLogin().pop();
+            }
+        };
```

Comment:
I assume that in the backport context we cannot do `Framework.login(username, username)` as in master?

Reply:
Indeed, in many cases we don't have the login module context and thus this method will fail without the whole feature improvement. I copy the same mechanism than [DummyLoginAs.java](https://github.com/nuxeo/nuxeo/blob/10.10/nuxeo-core/nuxeo-core-bulk/src/test/java/org/nuxeo/ecm/core/bulk/DummyLoginAs.java) from BAF.